Add member-function in edit-group

Tai-f opened the pull request:
# what
add check-box function in edit group

# why
to make groups include members by using this function

Commit:
create pull request

Commit:
Add "collection_check_boxes" in edit.html.haml

add "collection_check_boxes" in edit.html.haml
comment out some codes because they don’t need at this time

takebnn26 commented on app/views/groups/edit.html.haml:
```diff
@@ -8,14 +8,18 @@
+          =t.label{t.check_box + t.text}
     .chat-group-form__field.clearfix
       .chat-group-form__field--left
       .chat-group-form__field--right
```
groups/new.html.erbとgroups/edit.html.erbのファイルが似通っているので部分テンプレート化しましょう

Commit:
Add "end" to avoid errors

Commit:
Modify to the appropriate word

Modify to the appropriate word
 add “だ”

Commit:
Comment out some codes

Comment out some codes because they don’t need at this time

Commit:
Modify some codes to become same as edit.html.haml

Commit:
Modify some codes to become same as new.html.haml

Commit:
Generate the new file as partial

Generate the new file as partial for “new.html.haml” & “edit.html.haml”

Commit:
Duplicate the codes

Duplicate the codes which are same between "new.html.haml" &
"edit.html.haml" to group.html.haml

Commit:
Modify the partial file of "_groups"

Modify the partial file of "_groups"
it includes “if” method
it changed  some indentation to avoid errors

Commit:
Modify "edit.html.haml" to move some codes to the partial

Commit:
Modify "new.html.haml" to move some codes to the partial

FumiyaMori commented on app/views/groups/_group.html.haml:
```diff
@@ -0,0 +1,28 @@
+.chat-group-form__field.clearfix
+  .chat-group-form__field--left
+  .chat-group-form__field--right
+    = f.submit :Save, name:"commit", class: 'chat-group-form__action-btn', disable_with: 'Save'
```
このように部分テンプレートにする場合、form_forから部分テンプレートにしましょう。
form_forと、それに紐づくf.labelなどが異なるファイルに書かれてしまっているとわかりにくくなってしまいます。
またファイル名も_form.html.hamlなどのほうが適切かもしれません。

Tai-f replied:
了解しました

FumiyaMori commented on app/views/groups/_group.html.haml:
```diff
@@ -0,0 +1,28 @@
+.chat-group-form__field.clearfix
+  .chat-group-form__field--left
+    = f.label :グループ名, class: 'chat-group-form__label'
```
ラベルの機能が正しく使えているか確認してみてください

Tai-f replied:
正常に機能していると思われますが、ご指摘が何かご教授いただければ幸いです！

hiinaa replied:
labelは、フォーム部品とラベルを関係付けるために用いると思います。現状、viewに表示されるグループ名をクリックすると意図する動きにはなってないと思いますがどうでしょうか？

http://blog.willnet.in/entry/20080911/1221111749

FumiyaMori reviewed (commented):
コメント致しました

Commit:
Modify some codes in "new.html.haml

Delete form_for method due to move to the partial file

Commit:
Modify some codes in "edit.html.haml"

Delete some codes due to move to the partial file

Commit:
Modify some codes in "_group.html.haml"

Add “form_for”  method

Commit:
Rename the partial file

Rename the partial file due to use the appropriate name

FumiyaMori commented on app/views/groups/edit.html.haml:
```diff
@@ -1,22 +1,3 @@
-      .chat-group-form__field--left
-      .chat-group-form__field--right
-        = f.submit :Save, name:"commit", value: 'Save', class: 'chat-group-form__action-btn'
+  = render partial: 'group'
```
ファイル名を変更したのでそれにあわせましょう

FumiyaMori commented on app/views/groups/new.html.haml:
```diff
@@ -1,30 +1,5 @@
-      .chat-group-form__field--left
-      .chat-group-form__field--right
-        = f.submit :Save, name:'commit', value: 'Save', class: 'chat-group-form__action-btn', disable_with: 'Save'
+  = render partial: 'group'
```
ここもですね

hiinaa commented on app/views/groups/_group.html.haml:
```diff
@@ -0,0 +1,28 @@
+    / = f.label :チャットメンバー追加, class: 'chat-group-form__label'
+  .chat-group-form__field--right
+    .chat-group-form__search.clearfix
+      / = f.text_field :name, placeholder: "追加したいユーザー名を入力してください", id: 'user-search-field', class: 'chat-group-form__input'
```
コメントアウトは意図して行ってるのでしょうか？

FumiyaMori reviewed (commented):
コメント致しました

hiinaa commented on app/views/groups/edit.html.haml:
```diff
@@ -1,22 +1,4 @@
-      .chat-group-form__field--left
-      .chat-group-form__field--right
-        = f.submit :Save, name:"commit", value: 'Save', class: 'chat-group-form__action-btn'
+    = render partial: 'group', locals: {f: f}
```
他の人が見た時に、変数名が　f　だとわかりにくいかと思います。
細かいですが、わかりやすい変数名にしてあげましょう！

hiinaa commented on app/views/groups/new.html.haml:
```diff
@@ -1,30 +1,5 @@
-      .chat-group-form__field--left
-      .chat-group-form__field--right
-        = f.submit :Save, name:'commit', value: 'Save', class: 'chat-group-form__action-btn', disable_with: 'Save'
+    = render partial: 'group', locals: {f: f}
```
ここもですね！

Commit:
change the codes due to rename the partial file

change from “group to ”“form”

Commit:
Change the name due to rename the partial file

change from “group to ”“form”

Commit:
Comment out some codes

Comment out some codes because they don’t need at this time

Commit:
Cancel to comment out the codes

I cancel to comment out the codes  which do not influenced at this time

Commit:
Label makes association with the text field

kenchan0503 commented on app/views/groups/_form.html.haml:
```diff
@@ -0,0 +1,29 @@
+= form_for(@group) do|f|
```
細かいですがdoと|f|はスペースをお願いします

Tai-f replied:
了解です！

Commit:
Indents between "do and |f|" & "do and |t|"

add 1 space between “do and |f|” & “do and |t|”
